frontend/packages/pipelines-plugin/integration-tests/features/pipelines/create-from-builder-page.feature

```diff
@@ -130,3 +130,81 @@ Feature: Create the pipeline from builder page
+              And user selects the first task as "git-clone"
+              And user clicks on Add finally task
+              And user clicks on Select task
+              And user select "git-clone"
+              And user clicks on Add finally task again
+              And user clicks on Select task
+              And user selects "openshift-client"
+              And user clicks on Add finally task again
+              And user clicks on Select task
+              And user selects "kn"
+              And user fills task details
+              And user clicks on create
+             Then user sees finally tasks "git-clone", "openshift-client" and "kn" mentioned under "" section in the Pipeline details page
+
+
+        @regression, @manual
+        Scenario: When expression in the Pipeline Builder
+            Given user is at Pipeline Builder page
+              And user has chain of 4 tasks created in series
+             When user clicks on third task
+              And user navigates to When Expressions section
+              And user clicks on Add When Expressions
+             Then user sees a diamond shaped structure appear in front of third task
+              And user sees "Input", "Operator" and "Value" fields in When Expressions section
+              And user sees "Add Value", "Add When Expressions" and "Remove When Expressions" options
+
+
+        @regression
+        Scenario: Start pipeline with When expression in the Pipeline Builder
+            Given user is at Pipeline Builder page
+              And user has named pipeline as "pipeline-when-expression"
+              And user has tasks "git-cli", "git-clone" and "openshift-client" in series
+             When user clicks on third task
+              And user navigates to When Expressions section
+              And user clicks on Add When Expression
+              And user enters the input value as "$(tasks.git-clone.status)"
+              And user chooses the operator value "in" from the dropdown
+              And user enters the value as "Success"
+              And user clicks Create button on Pipeline Builder page
+             Then user will be redirected to Pipeline Details page with header name "pipeline-when-expression"
+
+
+        @regression, @manual
+        Scenario: Code assistance for referencing params in the Pipeline Builder
+            Given user is at Pipeline Builder page
+             When user clicks on Select task
+              And user select "git-clone"
+              And user enters pipeline name as "pipeline-code-assistance"
+              And user clicks on Add Parameter
+              And user adds Name as "git-url"
+              And user Default value as "https://github.com/sclorg/nodejs-ex.git"
+              And user clicks on Add Workspace and add name as "git-workspace"
+              And user clicks on "git-clone" task node
+              And user enters url under Parameters section "$(params.git-url)"
```

Comment:
Any plans to write more tests for the other variables? 

https://issues.redhat.com/browse/ODC-4277 showcases all the various code assistance params we plan to support (also available on https://tekton.dev/docs/pipelines/variables/)

Reply:
I've included one scenario each from the list of auto complete values from https://issues.redhat.com/browse/ODC-4277